### ISSUE TITLE
Support espresso/EarlGrey integration tests on engine PRs

### ISSUE THREAD

**Issue by xster** [flutter/flutter#32987]

**Comment by xster** [comment]
According to @dnfield and @tvolkert. The current state is:

- Android Junit+robolectric works today in presubmit. 
- Android firebase testlab + Android targeting espresso works today on LUCI in post submit
- Android espresso targeting Flutter is unblocked and WAI but not available yet. 
- iOS XCUnit test works today in postsubmit (since we can't build the iOS engine on cirrus)
- iOS simulator + earlgrey is fully supported in postsubmit
- iOS firebase + earlgrey is WIP by @dnfield 

Remaining 2019 infra work:

- Support Flutter targeting espresso APIs .
- Move XCUnit tests to pre-submit on iOS.

Did I get it right?

**Comment by xster** [comment]
Actually > Android firebase testlab + Android targeting espresso works today on LUCI in post submit
isn't true currently. 

That's the blocking work for phase 1 currently.

**Comment by xster** [comment]
I created some in-process AndroidJunit tests for now. 

Remaining blockers are to add that test to Cirrus firebase and add more espresso.

**Comment by adazh** [comment]
> Actually > Android firebase testlab + Android targeting espresso works today on LUCI in post submit
> isn't true currently.

Xiao, could you fill me in with more context or maybe point me to relevant issues? Especially - Is "espresso" equivalent to "robo tests" in your context? And - what are your testing goals/scenarios?

**Comment by xster** [comment]
It's just this bug tracking our infrastructure needs. 

Our needs will be tiered. Unit tests + robolectric will satisfy a a number of our unit testing and medium testing needs. We're unlikely to fulfill our JNI testing on them. 

The next tier would be JUnit tests running on actual Android environments. 

At the highest integration level, we'd want espresso tests to run against our engine commits and also in our https://github.com/flutter/samples repo. 

One example mentioned offline was to be able to bring up a native screen, press a button to show a Flutter screen. Press some other buttons to change some states in Flutter. Press pop to go back to native and assert that Flutter sends the state to native and the native state changed too.

**Comment by adazh** [comment]
Thanks! I don't think Robolectric today works with FlutterView as it requires loading native code on the host machine? But there's an effort on my team that is working on making native code working on the host.

**Comment by xster** [comment]
The espresso part is open sourced. This tracks the remaining work to support earl grey

**Comment by dnfield** [comment]
@jmagman - is this something the iOS team still wants?

/cc @stuartmorgan

**Comment by jmagman** [comment]
I don't think so, we have the scenario engine XCUITests running in presubmit on `Mac Unopt` now.
